applications/GeoMechanicsApplication/custom_processes/geo_integration_values_extrapolation_to_nodes_process.cpp

```diff
@@ -0,0 +1,242 @@
+{
+    InitializeAverageVariablesForElements();
+    InitializeVectorAndMatrixSizesOfVariables();
+}
+
+void GeoIntegrationValuesExtrapolationToNodesProcess::InitializeAverageVariablesForElements() const
+{
+    VariableUtils().SetNonHistoricalVariable(mrAverageVariable, 0.0, mrModelPart.Nodes());
+
+    block_for_each(mrModelPart.Elements(), [this](Element& rElement) {
+        if (rElement.IsActive()) {
+            for (auto& node : rElement.GetGeometry()) {
+                auto& node_var_to_update = node.GetValue(mrAverageVariable);
+                AtomicAdd(node_var_to_update, 1.0);
+            }
+        }
+    });
+
+    mrModelPart.GetCommunicator().AssembleNonHistoricalData(mrAverageVariable);
+}
+
+void GeoIntegrationValuesExtrapolationToNodesProcess::InitializeVectorAndMatrixSizesOfVariables()
+{
+    if (ModelPartContainsAtLeastOneElement()) {
+        auto r_first_element = mrModelPart.Elements().begin();
+
+        const GeometryData::IntegrationMethod this_integration_method =
+            r_first_element->GetIntegrationMethod();
+        const GeometryType::IntegrationPointsArrayType& integration_points =
+            r_first_element->GetGeometry().IntegrationPoints(this_integration_method);
+        const SizeType integration_points_number = integration_points.size();
+
+        const ProcessInfo& r_process_info = mrModelPart.GetProcessInfo();
+        InitializeSizesOfVectorVariables(*r_first_element, integration_points_number, r_process_info);
+        InitializeSizesOfMatrixVariables(*r_first_element, integration_points_number, r_process_info);
+    }
+}
+
+bool GeoIntegrationValuesExtrapolationToNodesProcess::ModelPartContainsAtLeastOneElement() const
+{
+    return !mrModelPart.Elements().empty();
+}
+
+void GeoIntegrationValuesExtrapolationToNodesProcess::InitializeSizesOfVectorVariables(
+    Element& rFirstElement, SizeType NumberOfIntegrationPoints, const ProcessInfo& rProcessInfo)
+{
+    for (const auto p_var : mVectorVariables) {
+        std::vector<Vector> values_on_integration_points(NumberOfIntegrationPoints);
+        rFirstElement.CalculateOnIntegrationPoints(*p_var, values_on_integration_points, rProcessInfo);
+        mSizesOfVectorVariables.insert({p_var, values_on_integration_points[0].size()});
+    }
+}
+
+void GeoIntegrationValuesExtrapolationToNodesProcess::InitializeSizesOfMatrixVariables(
+    Element& rFirstElement, SizeType NumberOfIntegrationPoints, const ProcessInfo& rProcessInfo)
+{
+    for (const auto p_var : mMatrixVariables) {
+        std::vector<Matrix> values_on_integration_points(NumberOfIntegrationPoints);
+        rFirstElement.CalculateOnIntegrationPoints(*p_var, values_on_integration_points, rProcessInfo);
+        std::pair<SizeType, SizeType> matrix_dimensions(values_on_integration_points[0].size1(),
+                                                        values_on_integration_points[0].size2());
+        mSizesOfMatrixVariables.try_emplace(p_var, matrix_dimensions);
```

Comment:
I would suggest to use `std::make_pair`:
```suggestion
        mSizesOfMatrixVariables.try_emplace(p_var, std::make_pair(values_on_integration_points[0].size1(), values_on_integration_points[0].size2()));
```

Reply:
Since the members have changed a bit, this is no longer necessary